### ISSUE TITLE
Allow to use custom serializaition in `localStorageCache` and `sessionStorageCache`

### ISSUE THREAD

**Issue by lenstr** [igorkamyshev/farfetched#304]
Hi! It would be super useful to have the ability to customize serialization used by `cache` adapters

### Use case 

We are using [tRPC](https://trpc.io/) with [superjson](https://github.com/blitz-js/superjson) as a custom transformer.
This allows transparently pass `Date` fields between client and server

```
export const api = createTRPCProxyClient<AppRouter>({
    transformer: superjson,
})
```

Each query are wrapped in `createQuery`
```
export const listQuery = createQuery({
    handler: api.list.query
})
```

Then we are trying to cache `listQuery`

```
cache(listQuery, {
    adapter: localStorageCache(),
})
```

Everything is working great, except instead of `Date` type in cached response we are getting `string` (e.g. `2023-04-03T07:22:51.627Z`), which is not the same response returned from a wrapped handler.

**Comment by igorkamyshev** [comment]
Hello! Sounds great, will be done in the next release 🫶

**Comment by igorkamyshev** [comment]
Done and will be released in #312
